okhttp-tests/src/test/java/okhttp3/HttpUrlTest.java

```diff
@@ -124,6 +122,12 @@
     assertEquals(null, base.newBuilder("ht.tp://b"));
   }
 
+  @Test public void redactedUrl() {
+    HttpUrl base = HttpUrl.parse("http://host/a/b#fragment");
```

Comment:
My bad for missing out on this case. Will remove the username and password from the redacted url if they exist.